x-pack/plugins/search_playground/public/hooks/use_source_indices_field.ts

```diff
@@ -5,24 +5,112 @@
  * 2.0.
  */
 
+import { useQuery } from '@tanstack/react-query';
+import { useController, useFormContext } from 'react-hook-form';
 import { IndexName } from '@elastic/elasticsearch/lib/api/types';
-import { useController } from 'react-hook-form';
-import { ChatFormFields } from '../types';
+import { useEffect, useState } from 'react';
+import { useKibana } from './use_kibana';
+import { APIRoutes, IndicesQuerySourceFields } from '../types';
+import { ChatForm, ChatFormFields } from '../types';
+import {
+  createQuery,
+  getDefaultQueryFields,
+  getDefaultSourceFields,
+  IndexFields,
+} from '../utils/create_query';
+
+export const getIndicesWithNoSourceFields = (
+  defaultSourceFields: IndexFields
+): string | undefined => {
+  const indices: string[] = [];
+  Object.keys(defaultSourceFields).forEach((index: string) => {
+    if (defaultSourceFields[index].length === 0) {
+      indices.push(index);
+    }
+  });
+
+  return indices.length === 0 ? undefined : indices.join();
+};
+
+export const useSourceIndicesFields = () => {
+  const { services } = useKibana();
+  const [loading, setLoading] = useState<boolean>(false);
+  const [noFieldsIndicesWarning, setNoFieldsIndicesWarning] = useState<string | null>(null);
+  const { resetField } = useFormContext<ChatForm>();
+
+  const {
+    field: { value: selectedIndices, onChange: onIndicesChange },
+  } = useController({
+    name: ChatFormFields.indices,
+  });
+
+  const {
+    field: { onChange: onElasticsearchQueryChange },
+  } = useController({
+    name: ChatFormFields.elasticsearchQuery,
+    defaultValue: {},
+  });
 
-export const useSourceIndicesField = () => {
   const {
-    field: { value: selectedIndices, onChange },
-  } = useController({ name: ChatFormFields.indices, defaultValue: [] });
+    field: { onChange: onSourceFieldsChange },
+  } = useController({
+    name: ChatFormFields.sourceFields,
+  });
+
+  const { data: fields } = useQuery({
+    enabled: selectedIndices.length > 0,
+    queryKey: ['fields', selectedIndices.toString()],
+    queryFn: async () => {
+      const response = await services.http.post<IndicesQuerySourceFields>(
+        APIRoutes.POST_QUERY_SOURCE_FIELDS,
+        {
+          body: JSON.stringify({ indices: selectedIndices }),
+        }
+      );
+      return response;
+    },
+  });
+
+  useEffect(() => {
+    if (fields) {
+      resetField(ChatFormFields.queryFields);
+
+      const defaultFields = getDefaultQueryFields(fields);
+      const defaultSourceFields = getDefaultSourceFields(fields);
+
+      const indicesWithNoSourceFields = getIndicesWithNoSourceFields(defaultSourceFields);
+
+      if (indicesWithNoSourceFields) {
```

Comment:
FYI @Samiul-TheSoccerFan - i've moved this warning state to the hook as now the code for dealing with fields / esQuery is done centrally.